cypress_test/integration_tests/desktop/calc/clipboard_spec.js

```diff
@@ -187,4 +187,36 @@ describe(['tagdesktop', 'tagnextcloud', 'tagproxy'], 'Calc clipboard tests.', fu
 
 		cy.cGet('#copy-paste-container').should('have.text', url);
 	});
+
+	it('Paste-Special', function () {
+		helper.setDummyClipboardForCopy();
+
+		calcHelper.clickOnFirstCell();
+		helper.typeIntoDocument('Something to copy paste.');
+		helper.typeIntoDocument('{enter}');
+		calcHelper.clickOnFirstCell();
+
+		cy.cGet('#map').rightclick(15, 15, { force: true });
+		cy.cGet('.on-the-fly-context-menu').should('be.visible');
+		cy.cGet('body').contains('.on-the-fly-context-menu .context-menu-item', 'Copy').should('be.visible');
+		cy.cGet('body').contains('.on-the-fly-context-menu .context-menu-item', 'Copy').click();
+
+		helper.typeIntoDocument('{rightArrow}');
+		helper.typeIntoDocument('{rightArrow}');
+
+		cy.cGet('#map').rightclick(210, 15, { force: true });
+		cy.cGet('.on-the-fly-context-menu').should('be.visible');
+		cy.cGet('body').contains('.on-the-fly-context-menu .context-menu-item', 'Paste Special').should('be.visible');
+		cy.cGet('body').contains('.on-the-fly-context-menu .context-menu-item', 'Paste Special').click();
+
+		cy.cGet('#modal-dialog-paste_special_dialog-box').should('be.visible');
+
+		cy.cGet('#modal-dialog-paste_special_dialog-box-yesbutton').should('be.visible');
+		cy.cGet('#modal-dialog-paste_special_dialog-box-yesbutton').click();
+
+		cy.cGet('#PasteSpecial').should('be.visible');
+		cy.cGet('#ok').click();
+
+		cy.cGet('#formulabar').should('contain.text', 'Something to copy paste.');
```

Comment:
Maybe assert here that #PasteSpecial is no longer visible? That was the primary problem visible to the user, that we had 2 dialogs, so clicking OK still left the user with a dialog.

Reply:
Ah thanks, let me add this soon, thanks for the review :)